Encourage :+1:'s for prioritisation

infinisil opened the pull request:
Adds a small text to each issue and the PR template to encourage people to use :+1:'s for issues they're also interested in. See https://github.com/NixOS/nix.dev/issues/359 for more information

This was done in the @NixOS/documentation-team

fricklerhandwerk commented on .github/ISSUE_TEMPLATE/bug_report.md:
````diff
@@ -38,3 +38,7 @@ Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
 output here
 ```
+
+### Prioritisation
+
+Add :+1: for [issues you find important](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).
````
We didn't decide on a subheading, and also I have no idea how it eluded us that it's "add x to y" and not "add x for y", unless *y* is the purpose of doing the addition.

```suggestion
### Priorities

Add :+1: to [issues you find important](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).
```

fricklerhandwerk replied:
![priorities](https://user-images.githubusercontent.com/6599296/205128490-b758f963-386a-4b28-82a1-cf6ce88cc904.jpg)

infinisil replied:
Not a big fan of this but done :)

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/2022-11-24-documentation-team-meeting-notes-17/23619/2

sternenseemann commented:
I think this has the potential to create unrealistic expectations. Issues get resolved because some *individual* person cares enough and has the ability to fix it. This may or may not correlate with a popular demand for the issue being fixed.

This fact can possibly create bad interactions in both directions:

* “Casual” users get frustrated because an issue that has a lot of upvotes doesn't actually get prioritised over others and continues to not be fixed.
* Volunteer contributors get frustrated because they feel they have to work on upvoted issues that are not fun to fix for them.

Additionally, the following questions are unanswered for me:

* Does having this hint everywhere make a meaningful difference or do people already naturally use emoji reactions?
* Will the emoji sorting be useful? Here I can vividly imagine that the top of the list will be cluttered by very hard to fix / large scale issues or popular ideas (like a `nixFlakes` attribute) that don't make sense from a project perspective.

SuperSandro2000 commented:
I would like to see explicitly mentioned that we mean reactions **and** not comments.

fricklerhandwerk commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -38,3 +38,7 @@ Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
+
+### Priorities
+
+Add :+1: to [issues you find important](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).
```
@SuperSandro2000 is this what you have in mind?

```suggestion
Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc
```

fricklerhandwerk commented:
@sternenseemann your concerns are valid, and the complement to such a convention should be transparency on the maintainer( team)s' side and clear communication towards volunteers.

The idea is to give guidance on what would be worthwhile to work on. Right now this is hard to judge for beginners and intermediates, because there is no overview on what users value.

> Does having this hint everywhere make a meaningful difference or do people already naturally use emoji reactions?

Yes, I think it will make a difference making the existing convention explicit. Right now "the usual suspects" do this, but the community is growing quickly, and most beginners will stay passive unless provided with instructions.

>  Will the emoji sorting be useful? Here I can vividly imagine that the top of the list will be cluttered by very hard to fix / large scale issues or popular ideas (like a `nixFlakes` attribute) that don't make sense from a project perspective.

That also needs to be complemented by clear communication, and e.g. closing issues which won't be addressed no matter what.

fricklerhandwerk commented on .github/ISSUE_TEMPLATE/bug_report.md:
````diff
@@ -38,3 +38,7 @@ Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.
 output here
 ```
+
+### Priorities
````
@sternenseemann one could just remove the word "priorities" to avoid unwanted associations.

This is what I thought initially anyway, it should just be a note, a nudge to action.

```suggestion
---
```

sternenseemann replied:
Yeah, still think priorities is misleading.

sternenseemann replied:
Also this should be in a comment and not show up in the actual posted issue (like with the PR template).

fricklerhandwerk replied:
Oh but the point is exactly for readers to see it. Otherwise only issue authors would be informed of the mechanism.

fricklerhandwerk commented:
@sternenseemann @SuperSandro2000 anything to add? Can we merge this?

infinisil commented:
I updated all the templates to use the new format and squashed the commits.

Note that there's essentially no cost associated with this change, and the benefit we could get from this change could be very significant.

katexochen reviewed (approved):
Got here due to the prioritization of this PR. ;)
I think it is worth trying.

infinisil committed:
Encourage +1's for prioritisation

Adds a small text to each issue and the PR template to encourage people to use
:+1:'s for issues they're also interested in. See
https://github.com/NixOS/nix.dev/issues/359 for more information

Co-Authored-By: Valentin Gagarin <valentin.gagarin@tweag.io>

infinisil commented:
It sure does help! With these 8 upvotes I'm also confident enough to merge this myself :smile:
